Src/GoogleApis.Auth/OAuth2/AuthorizationCodeInstalledApp.cs

```diff
@@ -88,7 +87,18 @@ public ICodeReceiver CodeReceiver
                     taskCancellationToken).ConfigureAwait(false);
             }
 
-            return new UserCredential(flow, userId, token);
+            return new UserCredential(Flow, userId, token);
+        }
+
+        /// <summary>
+        /// Determines the need for retrieval of a new authorization code, based on the given token and the 
+        /// authorization code flow.
+        /// </summary>
+        public bool ShouldRequestAuthorizationCode(TokenResponse token)
+        {
+            // If the flow includes a parameter that requires a new token, if the stored token is null or it doesn't
+            // have a refresh token and the access token is expired we need to retrieve a new authorization code.
+            return Flow.ShouldForceTokenRetrieval() || token == null || (token.RefreshToken == null && token.IsExpired(flow.Clock));

```

Comment:
Just verify that the line length is no more than 120 :) It's hard to verify that using this viewer...